### DISCUSSION TITLE
security: CVE-2023-49569 - go-git/v5

### DISCUSSION THREAD

**Discussion by cmontemuino** [dexidp/dex#3272] [General]
We've observed (Trivy scan) ghcr.io/dexidp/dex:v2.37.0 includes critical vulnerability CVE-2023-4956 in package `github.com/go-git/go-git/v5`.

- https://github.com/advisories/GHSA-449p-3h89-pw88
- https://nvd.nist.gov/vuln/detail/CVE-2023-4956

The vulnerability does not come from dex binary, but `gotemplate`. I've filed an issue in their repo and proposed a fix: https://github.com/hairyhenderson/gomplate/issues/1960

When it's accepted and released, then a new version of `dexidp/dex` might be created.

Note: it makes all images that depend on dexidp/dex contain that critical vulnerability.

**Comment by nabokihms** [comment]
Thanks for mentioning it, @cmontemuino.  👍

**Reply by cmontemuino** [reply]
@nabokihms according to gotemplate's maintainer, the reported vulnerability is not applicable, so this issue might be closed:

> This particular vulnerability is not applicable to gomplate's use of go-git, as gomplate only uses the in-memory filesystem when communicating with remote git servers:
>> Applications using [BoundOS](https://pkg.go.dev/github.com/go-git/go-billy/v5/osfs#BoundOS) or in-memory filesystems are not affected by this issue.

https://github.com/hairyhenderson/gomplate/issues/1960#issuecomment-1889261549

**Comment by dlukasiewicz** [comment]
Hey, seems like the issue on go is already closed, pull request is merged. Can You guys release new version with this package fixed? 
"When it's accepted and released, then a new version of dexidp/dex might be created." i can still see that v2.37.0 is the newest one 

Best Regards :)  @cmontemuino  @nabokihms

**Reply by nabokihms** [reply]
Opened a PR https://github.com/dexidp/dex/pull/3293
